### PR TITLE
Parent child services

### PR DESCRIPTION
**Pull request by jeremyrickard**
This PR brings over our changes for supporting parent child services and a refactor of the SQLDDB service to support three distinct services:

* All In One (server and db)
* Server Only (only the server VM, must define an alias)
* VM only (requires a parent declaration)

### CODE REVIEW

File: pkg/api/deprovision.go [hunk 1]
```diff
@@ -119,21 +120,46 @@ func (s *server) deprovision(w http.ResponseWriter, r *http.Request) {
...
+		)
+	}
 	if err = s.asyncEngine.SubmitTask(task); err != nil {
 		logFields["step"] = firstStepName
 		logFields["error"] = err
 		log.WithFields(logFields).Error(
 			"deprovisioning error: error submitting deprovisioning task",
 		)
 		s.writeResponse(w, http.StatusInternalServerError, responseEmptyJSON)
-		return
```
Comment (krancour) [inline pkg/api/deprovision.go]: I don't think this return statement should have been removed.

Comment (jeremyrickard) [reply pkg/api/deprovision.go]: Weird! I'll add that back. I'll scan through the rest of these too

File: pkg/service/catalog.go [hunk 2]
```diff
@@ -39,9 +41,12 @@ type Service interface {
 	ToJSON() ([]byte, error)
 	GetID() string
 	GetName() string
+	GetBindable() bool
```
Comment (krancour) [inline pkg/service/catalog.go]: Can we change this to `IsBindable()`? That seems more conventional when you're asking a yes/no question and expecting a boolean response.

File: pkg/service/instance.go [hunk 3]
```diff
@@ -22,6 +24,7 @@ type Instance struct {
 	StatusReason                    string                 `json:"statusReason"`
 	Location                        string                 `json:"location"`
 	ResourceGroup                   string                 `json:"resourceGroup"`
+	ParentAlias                     string                 `json:"parentAlias"`
```
Comment (krancour) [inline pkg/service/instance.go]: Can we move this up above `Parent`... or else move `Parent` down below this? I just want to keep them together, so it's more obvious they're related.

**Comment by krancour** [comment]
LGTM. Will merge pending CI and some manual e2e on my end.

**Pull request merged**